presto-main/src/main/java/com/facebook/presto/server/TaskInfoResource.java

```diff
@@ -63,10 +95,58 @@ public TaskInfo getTaskInfo(@PathParam("taskId") TaskId taskId)
                     }
                 }
             }
-            throw new NotFoundException("TaskInfo not found for task id " + taskId.toString());
+            throw new NotFoundException("TaskInfo not found for task id " + taskId);
         }
         catch (Exception e) {
             throw new NotFoundException(e);
         }
     }
+
+    @GET
+    @Path("{taskId}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public void getTaskInfo(@PathParam("taskId") TaskId taskId,
+            @QueryParam("includeLocalQueryOnly") @DefaultValue("false") boolean includeLocalQueryOnly,
+            @HeaderParam(X_FORWARDED_PROTO) String xForwardedProto,
+            @Context UriInfo uriInfo,
+            @Context HttpServletRequest servletRequest,
+            @Suspended AsyncResponse asyncResponse)
+            throws WebApplicationException
+    {
+        try {
+            if (resourceManagerEnabled && !dispatchManager.isQueryPresent(taskId.getQueryId()) && !includeLocalQueryOnly) {
+                proxyTaskInfo(servletRequest, asyncResponse, xForwardedProto, uriInfo);
+            }
+            else {
+                asyncResponse.resume(Response.ok(getTaskInfo(taskId)).build());
+            }
+        }
+        catch (NotFoundException e) {
+            asyncResponse.resume(Response.status(NOT_FOUND).build());
+        }
```

Comment:
We proxy to the RM if:
* RM is enabled,
* We don't have the query on this coordinator, and
* This request isn't a proxied call itself, meaning we shouldn't proxy it again.

Otherwise, the request is regularly handled (by returning the output of getTaskInfo).

If we hit a `NotFoundException` while calling `getTaskInfo` then we return 404.

Reply:
when do we hit a `NotFoundException` . Do we need to handle it separately.. Wouldn't the Jetty filters take care of returning a 404.

Reply:
Fixed: Let Jetty handle returning a 404.